app/src/components/events/Recommend.tsx

```diff
@@ -0,0 +1,87 @@
+import { QueryDocumentSnapshot, collection, doc, getDocs, query, setDoc, where } from "firebase/firestore";
+import { db } from "../../config/firebase-config";
+import { useEffect, useState } from "react";
+import data from "./events.json"
+
+export function Recommend() {
+    const [events, setEvents] = useState<QueryDocumentSnapshot[]>();
+    const [histRating, setHistRating] = useState<number | undefined>();
+    const [value, setValue] = useState<number | undefined>();
+
+    function averageRating(collection: QueryDocumentSnapshot[] | undefined) {
+        var sum = 0;
+        
+        if (collection) {
+            collection.forEach(document => {
+                sum += average(document.data().ratings);
+              });
+            
+            setHistRating(sum / collection.length);
+            return (sum / collection.length);
+        } else {
+            return 0;
+        }
+      }
+    
+    const average = (array: number[]) => {
+        var sum = 0;
+    
+        array.forEach(element => {
+          sum += element;
+        });
+    
+        return (sum / array.length);
```

Comment:
are we guarding against an empty array that makes this a division over 0?

Reply:
thank you for pointing that out, its fixed now

Reply:
Actually it isnt necessary because the only place that this function is called, its within the condition that the ratings array has value and therefore nonzero length